esti/s3_gateway_test.go

```diff
@@ -658,30 +658,27 @@ func TestS3ReadObjectRedirect(t *testing.T) {
 	})
 }
 
+func createS3Client(endpoint string, t *testing.T) *s3.Client {
```

Comment:
Just as thought,
There are several places who use `testutil.SetupTestS3Client` to change the endpoint (as you already noticed).
Suggest to put this func under the testutil package and use it in all the places where `testutil.SetupTestS3Client` is called.

Reply:
I would prefer _not_ to do this.  `viper` state is, by definition, **global**.  So I want all uses of viper to appear in the test, rather than in a "don't worry about it" setup function.